pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -339,23 +339,29 @@ func updateCEPUID(scopedLog *logrus.Entry, e *endpoint.Endpoint, localCEP *ciliu
 	// If the Endpoint already owns the CEP (by holding the matching CEP UID reference) then we don't have to
 	// worry about other ownership checks.
 	//
-	// This will cover cases such as if the NodeIP changes (as with a reboot). In which case we can safely
-	// take ownership and overwrite the CEPs status.
+	// This will cover cases such as if the NodeIP changes (as with a reboot).
+	// In which case we can safely take ownership and overwrite the CEPs
+	// status. However if the cilium endpoints were previously checkpointed
+	// into tmpfs this check will fail and we will rely on the next check to
+	// prevent us from hijacking CEPs.
 	cepUID := e.GetCiliumEndpointUID()
 	if cepUID == localCEP.UID {
 		return nil
 	}
 
-	var nodeIP string
-	if netStatus := localCEP.Status.Networking; netStatus == nil {
-		return fmt.Errorf("endpoint sync cannot take ownership of CEP that has no nodeIP status")
-	} else {
-		nodeIP = netStatus.NodeIP
-	}
-
 	// We do not want to take ownership of CEPs created on other Nodes.
-	if nodeIP != node.GetCiliumEndpointNodeIP() {
-		return fmt.Errorf("endpoint sync cannot take ownership of CEP that is not local (%q)", nodeIP)
+	// However we can't directly compare the CEP node ip with the node, because
+	// the node ip can change, orphaning the CEP. So we retrieve the pod for
+	// the CEP and compare its node IP with that of the node. The kubelet on
+	// this node will update the pod object appropriately, allowing this check
+	// to eventually go through.
+	// The intent here is to check if a given pod is running on the same node
+	// this cilium is running on before taking over its CEP.
+	nodeIP := node.GetCiliumEndpointNodeIP()
```

Comment:
this function does not have any unit test, can we add one to cover this bug and avoid regression?

Reply:
Yep, will add by EOD. Was just waiting for first pass comments.

Reply:
Fixed and added unittests.